Added table for removed properties (#936)
cap-js/docs: java/migration.md

| `cds.remote.services.<key>.http.csrf.enabled` | `true` | `false` | Most APIs don't require CSRF tokens. |

> ### Changed Properties

Reviewer: I think we can say they are removed.

The following table gives an overview about the changed properties:

> | old Property | new Property | Explaination |

Reviewer: ```suggestion
| Removed Property | Replacement | Explanation |
```

| `cds.data-source.csvSingleChangeset` | `cds.data-source.csv.singleChangeset` | |
| `cds.remote.<key>.destination.type` | `cds.remote.services.<key>.type` | |
| `cds.remote.<key>.destination.suffix` | `cds.remote.services.<key>.http.suffix` | |
| `cds.remote.<key>.destination.service` | `cds.remote.services.<key>.http.service` | |
| `cds.remote.<key>.destination.headers` | `cds.remote.services.<key>.http.headers` | |
| `cds.remote.<key>.destination.queries` | `cds.remote.services.<key>.http.queries` | |
| `cds.sql.search.useLocalizedView` | `cds.sql.search.model` | |
| `cds.identity.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` | |
| `cds.xsuaa.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` | |
| `cds.odataV4.apply.inCqn.enabled` | `cds.odataV4.apply.transformations.enabled` | |
| `cds.multiTenancy.healthCheck.intervalMillis` | `cds.multiTenancy.healthCheck.interval` | |
| `cds.messaging.services.<key>.outbox.persistent.enabled` | `cds.messaging.services.<key>.outbox.name` | |
| `cds.auditlog.outbox.persistent.enabled` | `cds.auditlog.outbox.name` | |
| `cds.outbox.persistent` | `cds.outbox.services.<key>` | |

> | `cds.sql.supportedLocales` | none | All locales are supported by default for localized entities, as session variables can now be leveraged on all databases |
> | `cds.security.mock.users.<key>.unrestricted` | none | to be filled |

Reviewer: ```suggestion
| `cds.sql.supportedLocales` | | All locales are supported by default for localized entities, as session variables can now be leveraged on all databases |
| `cds.security.mock.users.<key>.unrestricted` | | Special handling of unrestricted attributes has been removed, in favor of [explicit modelling](../guides/security/authorization#unrestricted-xsuaa-attributes). |
```

### Replaced Properties

> Some deprecated properties are replaced with new properties. The following table gives an overview about the replaced properties:

Reviewer: Maybe we shouldn't distignuish between replaced properties and removed properties. Basically all properties are removed. But some have an existing replacement, that is already available today. Currently it sounds like the replacement will only be introduced with 3.0.

Author: ok, changed the text.

| `cds.data-source.csvSingleChangeset` | `cds.data-source.csv.singleChangeset` |
| `cds.remote.<key>.destination.type` | `cds.remote.services.<key>.type` |
| `cds.remote.<key>.destination.suffix` | `cds.remote.services.<key>.http.suffix` |
| `cds.remote.<key>.destination.service` | `cds.remote.services.<key>.http.service` |
| `cds.remote.<key>.destination.headers` | `cds.remote.services.<key>.http.headers` |
| `cds.remote.<key>.destination.queries` | `cds.remote.services.<key>.http.queries` |
| `cds.sql.search.useLocalizedView` | `cds.sql.search.model` |
| `cds.identity.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` |
| `cds.xsuaa.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` |
| `cds.odataV4.apply.inCqn.enabled` | `cds.odataV4.apply.transformations.enabled` |
| `cds.multiTenancy.healthCheck.intervalMillis` | `cds.multiTenancy.healthCheck.interval` |
| `cds.messaging.services.<key>.outbox.persistent.enabled` | `cds.messaging.services.<key>.outbox.name` |
| `cds.auditlog.outbox.persistent.enabled` | `cds.auditlog.outbox.name` |
| `cds.outbox.persistent` | `cds.outbox.services.<key>` |

### Removed Properties

> Some deprecated properties are completely removed without a replacement:

Reviewer: We should also explain WHY there are no replacements for these properties:

- supportedLocales: All locales are supported by default for localized entities, as session variables can now be leveraged on all databases
- unrestricted stuff: Maybe this is worth a dedicated chapter, as we also remove the methods there. We can link to the section in CAPire how to handle unrestricted attributes: https://pages.github.tools.sap/cap/docs/guides/security/authorization#unrestricted-xsuaa-attributes

Author: merged both props into the table

| `cds.remote.services.<key>.http.csrf.enabled` | `true` | `false` | Most APIs don't require CSRF tokens. |

### Removed Properties

The following table gives an overview about the removed properties:

| Removed Property | Replacement | Explanation |
| --- | --- | --- |

> | `cds.data-source.csvInitializationMode` | `cds.data-source.csv.initializationMode` | |
> | `cds.data-source.csvFileSuffix` | `cds.data-source.csv.fileSuffix` | |
> | `cds.data-source.csvPaths` | `cds.data-source.csv.paths` | |
> | `cds.data-source.csvSingleChangeset` | `cds.data-source.csv.singleChangeset` | |

Reviewer: ```suggestion
| `cds.dataSource.csvInitializationMode` | `cds.dataSource.csv.initializationMode` | |
| `cds.dataSource.csvFileSuffix` | `cds.dataSource.csv.fileSuffix` | |
| `cds.dataSource.csvPaths` | `cds.dataSource.csv.paths` | |
| `cds.dataSource.csvSingleChangeset` | `cds.dataSource.csv.singleChangeset` | |
```

| `cds.remote.<key>.destination.queries` | `cds.remote.services.<key>.http.queries` | |
| `cds.sql.search.useLocalizedView` | `cds.sql.search.model` | |
| `cds.identity.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` | |
| `cds.xsuaa.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` | |
| `cds.odataV4.apply.inCqn.enabled` | `cds.odataV4.apply.transformations.enabled` | |
| `cds.multiTenancy.healthCheck.intervalMillis` | `cds.multiTenancy.healthCheck.interval` | |
| `cds.messaging.services.<key>.outbox.persistent.enabled` | `cds.messaging.services.<key>.outbox.name` | |
| `cds.auditlog.outbox.persistent.enabled` | `cds.auditlog.outbox.name` | |
| `cds.outbox.persistent` | `cds.outbox.services.<key>` | |
| `cds.sql.supportedLocales` | | All locales are supported by default for localized entities, as session variables can now be leveraged on all databases. |
| `cds.security.mock.users.<key>.unrestricted` | | Special handling of unrestricted attributes has been removed, in favor of [explicit modelling](../guides/security/authorization#unrestricted-xsuaa-attributes). |

> | `cds.multiTenancy.mtxs.enabled` | - | MTXS is enabled [by default](#removed-mtx-classic-support). |
> | `cds.multiTenancy.compatibility.enabled` | - | MtSubscriptionService API [has been removed](#removed-mtx-classic-support) and compatibility mode is no longer available. |
> | `cds.multiTenancy.security.deploymentScope` | - | HTTP-based tenant upgrade endpoints [have been removed](#removed-mtx-classic-support). |

Reviewer: ```suggestion
| `cds.multiTenancy.mtxs.enabled` | | MTXS is enabled [by default](#removed-mtx-classic-support). |
| `cds.multiTenancy.compatibility.enabled` | | MtSubscriptionService API [has been removed](#removed-mtx-classic-support) and compatibility mode is no longer available. |
| `cds.multiTenancy.security.deploymentScope` | | HTTP-based tenant upgrade endpoints [have been removed](#removed-mtx-classic-support). |
```
